### PR TITLE
pkg/tz(ticdc): use correct ctx to get timezone

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/8798

### What is changed and how it works?
- Fix a bug that unescape the timezone value twice
- Fix a bug we use the wrong ctx when we get the timezone.
- Warning when users use time-zone="" in sink-uri.
- Warning when users do not set the timezone in sink-uri

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
     - https://github.com/pingcap/tiflow/pull/8799#issuecomment-1512417945
     - https://github.com/pingcap/tiflow/pull/8799#issuecomment-1512438022

##### Will it cause performance regression or break compatibility?
None
##### Do you need to update user documentation, design documentation or monitoring documentation?
None

### Release note 

```release-note
Fix the bug that TiCDC always uses `UTC` when creating changefeed
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
Test it locally to make sure we use the correct timezone from the ctx.
1. install the MySQL with homebrew, it would use the system timezone as MySQL timezone
2. start upstream tidb cluster, build and ticdc with this PR
3. start ticdc server without timezone config(it would use system)
4. try to create changefeed, it would use the Asia/Shanghai instead of UTC.
```console
➜  bin git:(rustin-patch-timezone) ✗ ./cdc cli changefeed create --sink-uri="mysql://root:123456@127.0.0.1:3306" -c "test1"
Error: [CDC:ErrMySQLConnectionError]fail to open MySQL connection: Error 1298 (HY000): Unknown or incorrect time zone: 'Asia/Shanghai'
```

**Comment by hi-rustin** [comment]
Test it locally to make sure it will bail out when the changefeed timezone is different from server.
1. install the MySQL with homebrew, it would use the system timezone as MySQL timezone
2. start upstream tidb cluster, build and ticdc with this PR
3. start ticdc server without timezone config(it would use system)
4. try to create changefeed, it would bail out an error.
```console
➜  bin git:(rustin-patch-timezone) ./cdc cli changefeed create --sink-uri="mysql://root:123456@127.0.0.1:3306?time-zone=UTC" -c "test3"
Error: [CDC:ErrMySQLInvalidConfig]MySQL config invalid: the timezone of the TiCDC server and the sink-uri are inconsistent. TiCDC server timezone: Asia/Shanghai, sink-uri timezone: UTC. Please make sure that the timezone of the TiCDC server, sink-uri and the downstream database are consistent.
```

File: pkg/sink/mysql/config.go [hunk 1]
```diff
@@ -330,28 +331,48 @@ func getSafeMode(values url.Values, safeMode *bool) error {
...
 	if err != nil {
 		return cerror.WrapError(cerror.ErrMySQLInvalidConfig, err)
 	}
-	_, err = time.LoadLocation(value)
-	if err != nil {
-		return cerror.WrapError(cerror.ErrMySQLInvalidConfig, err)
+	*timezone = fmt.Sprintf(`"%s"`, changefeedTimezone.String())
+	// We need to check whether the timezone of the TiCDC server and the sink-uri are consistent.
+	// If they are inconsistent, it may cause the data to be inconsistent.
+	if changefeedTimezone.String() != serverTimezone.String() {
```
Comment (nongfushanquan) [inline pkg/sink/mysql/config.go:367]: If the timezone sink in the uri is set by the user , the timezone of the server should be ignored

**Review by nongfushanquan — APPROVED** [review]

File: pkg/sink/mysql/config.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/sink/mysql/config.go:367]: We can do it in the future. Because we need to update the init order for the mounter.

**Review by amyangfei — APPROVED** [review]

**Pull request merged**